app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinderTile.java

```diff
@@ -1,36 +1,50 @@
 package com.concordia.mcga.utilities.pathfinding;
 
-import android.util.Log;
+import android.support.annotation.NonNull;
 
-import org.apache.commons.lang3.builder.EqualsBuilder;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-import org.json.JSONException;
-import org.json.JSONObject;
+/**
+ * Object acts as a wrapper for {@link IndoorMapTile}. Holds
+ * additional information about the tiles that are specifically used
+ * by the pathfinder algorithms
+ */
+public class PathFinderTile implements Comparable<PathFinderTile> {
 
-public class PathFinderTile {
+    enum Type {START, DESTINATION}
 
-    public static final PathFinderTile MAX_COST = new PathFinderTile(Integer.MIN_VALUE, Integer.MIN_VALUE);
+    public static final PathFinderTile MAX_COST = new PathFinderTile();
 
     static {
         MAX_COST.setDistFromEnd(Integer.MAX_VALUE);
     }
 
+    private IndoorMapTile indoorMapTile;
     private Type tileType;
     private int distFromStart;
     private int distFromEnd;
-    private int coordinateX;
-    private int coordinateY;
     private PathFinderTile parent;
 
-    public PathFinderTile(int x, int y) {
-        this.coordinateX = x;
-        this.coordinateY = y;
+    private PathFinderTile(){}
+
+    public PathFinderTile(IndoorMapTile indoorMapTile) {
+        this.indoorMapTile = indoorMapTile;
+    }
+
+    public Type getTileType() {
+        return tileType;
     }
 
     public void setTileType(Type tileType) {
         this.tileType = tileType;
     }
 
+    public IndoorMapTile getIndoorMapTile() {
+        return indoorMapTile;
+    }
+
+    public void setIndoorMapTile(IndoorMapTile indoorMapTile) {
+        this.indoorMapTile = indoorMapTile;
+    }
+
     public int getCost() {
```

Comment:
Well maybe... but the distance from the start is actually being calculated, but the distance to destination is an approximation by design; it is a heuristic to make the algorithm efficient rather than accurate.

This is the way the A* pathfinding algorithm works.